perfkitbenchmarker/providers/ibmcloud/ibmcloud_virtual_machine.py

```diff
@@ -0,0 +1,599 @@
+        logging.info('account_id: %s', account.name)
+        if FLAGS.ibmcloud_datavol_encryption_key:  # if this flag is set
+          self.data_encryption_key = account.enckey
+          logging.info('KP key to use, data: %s', self.data_encryption_key)
+        if FLAGS.ibmcloud_bootvol_encryption_key:  # use same key as data
+          self.boot_encryption_key = account.enckey
+          logging.info('KP key to use, boot: %s', self.boot_encryption_key)
+
+    if not account.name or not account.apikey:
+      raise errors.Config.InvalidValue(
+          'PerfKit Benchmarker on IBM Cloud requires that the '
+          'environment variables IBMCLOUD_ACCOUNT_ID and IBMCLOUD_APIKEY '
+          'are correctly set.')
+    ibm.IbmAPICommand.ibmcloud_account_id = account.name
+    ibm.IbmAPICommand.ibmcloud_apikey = account.apikey
+    IbmCloudVirtualMachine.ibmcloud_account_id = account.name
+    IbmCloudVirtualMachine.ibmcloud_apikey = account.apikey
+
+  def _CreateInstance(self):
+    """Creates IBM Cloud VM instance."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags.update({
+      'name': self.name,
+      'imageid': self.imageid,
+      'profile': self.profile,
+      'vpcid': self.vpcid,
+      'subnet': self.subnet,
+      'key': self.key,
+      'zone': self.zone
+      })
+    cmd.user_data = self.user_data
+    if self.boot_volume_size > 0:
+      cmd.flags['capacity'] = self.boot_volume_size
+    cmd.flags['iops'] = self.boot_volume_iops
+    if self.boot_encryption_key:
+        cmd.flags['encryption_key'] = self.boot_encryption_key
+    logging.info('Creating instance, flags: %s', cmd.flags)
+    resp = json.loads(cmd.CreateInstance())
+    if 'id' not in resp:
+      raise errors.Error(f'IBM Cloud ERROR: Failed to create instance: {resp}')
+    self.vmid = resp['id']
+    logging.info('Instance created, id: %s', self.vmid)
+    logging.info('Waiting for instance to start, id: %s', self.vmid)
+    cmd.flags['instanceid'] = self.vmid
+    status = cmd.InstanceStatus()
+    self.instance_start_failed = False
+    assert status == ibm.States.RUNNING
+    if status != ibm.States.RUNNING:
+      logging.error('Instance start failed, status: %s', status)
+      self.instance_start_failed = True
+    self.vm_started = not self.instance_start_failed
+    logging.info('Instance %s status %s', self.vmid, status)
+
+  def _StartInstance(self):
+    """Starts a IBM Cloud VM instance."""
+    cmd = ibm.IbmAPICommand(self)
+    cmd.flags['instanceid'] = self.vmid
+    status = cmd.InstanceStart()
+    logging.info('start_instance_poll: last status is %s', status)
+    self.instance_start_failed = False
+    assert status == ibm.States.RUNNING
+    if status != ibm.States.RUNNING:
+      logging.error('Instance start failed, status: %s', status)
+      self.instance_start_failed = True
+    self.vm_started = not self.instance_start_failed
+
+  def _CreateFip(self, name: str):
+    """Creates a VNIC in a IBM Cloud VM instance."""
```

Comment:
This looks better suited for the ibmcloud_network.py file

Reply:
moved.